services-custom/dynamodb-enhanced/src/main/java/software/amazon/awssdk/enhanced/dynamodb/internal/DynamoDBEnhancedRequestConfiguration.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.enhanced.dynamodb.internal;
+
+import software.amazon.awssdk.annotations.SdkPublicApi;
+import software.amazon.awssdk.enhanced.dynamodb.mapper.AttributeMapping;
+
+/**
+ * Configuration Object to define behaviour of DynamoDB operations.
+ */
+@SdkPublicApi
+public class DynamoDBEnhancedRequestConfiguration {
+    private final AttributeMapping attributeMapping;
+
+    public DynamoDBEnhancedRequestConfiguration(AttributeMapping attributeMapping) {
+        this.attributeMapping = attributeMapping;
+    }
```

Comment:
Because this is a public API, can we add a Builder for backwards compatibility

Reply:
+1 Let's use builder pattern instead of the ctor. Let's follow the config class convention. https://github.com/aws/aws-sdk-java-v2/blob/master/docs/design/ClientConfiguration.md